### ISSUE TITLE
BadImageFormatException

### ISSUE THREAD

**Issue by sustundag** [getgauge/gauge-dotnet#85]
Hi,
I recently created an issue with Sql access. 
[Issue 80](https://github.com/getgauge/gauge-dotnet/issues/80)
Firstly, Thank you for your support on the subject. 
Demo application which I worked is same with on issue 80.

As you suggessed, I set the runtime like below.
dotnet.properties file:
GAUGE_CSHARP_PROJECT_CONFIG=release --runtime=win-x64

And then, I defined a valid connection string to my test sql server.
`SqlConnection ss=new SqlConnection("Server=...,1433;Initial Catalog=gaugetest;Persist Security Info=False; User ID=...;Password=...; MultipleActiveResultSets=False; Encrypt=True; TrustServerCertificate=False; Connection Timeout=30;");`
`ss.Open();`

When I run the specs, I am getting BadImageFormatException
Here is the stacktrace.

> Unhandled exception.
> System.BadImageFormatException: Bad IL format.
> at System.Runtime.Loader.AssemblyLoadContext.InternalLoad(ReadOnlySpan`1 arrAssembly, ReadOnlySpan`1 arrSymbols)
> at System.Runtime.Loader.AssemblyLoadContext.LoadFromStream(Stream assembly, Stream assemblySymbols)
> at System.Runtime.Loader.AssemblyLoadContext.LoadFromStream(Stream assembly)
> at Gauge.Dotnet.Wrappers.AssemblyWrapper.LoadFrom(String location) in /home/runner/work/gauge-dotnet/gauge-dotnet/src/Wrappers/AssemblyWrapper.cs:line 43
> at Gauge.Dotnet.AssemblyLoader.ScanAndLoad(String path) in /home/runner/work/gauge-dotnet/gauge-dotnet/src/AssemblyLoader.cs:line 139
> at Gauge.Dotnet.AssemblyLoader..ctor(IAssemblyWrapper assemblyWrapper, IEnumerable`1 assemblyLocations, IReflectionWrapper reflectionWrapper, IActivatorWrapper activatorWrapper, IStepRegistry registry) 
> in /home/runner/work/gauge-dotnet/gauge-dotnet/src/AssemblyLoader.cs:line 48
> at Gauge.Dotnet.GaugeListener.StartServer() in /home/runner/work/gauge-dotnet/gauge-dotnet/src/GaugeListener.cs:line 40
> at Gauge.Dotnet.StartCommand.Execute() in /home/runner/work/gauge-dotnet/gauge-dotnet/src/StartCommand.cs:line 46
> at Gauge.Dotnet.Program.Main(String[] args) in /home/runner/work/gauge-dotnet/gauge-dotnet/src/Program.cs:line 38
> Error occurred while waiting for runner process to finish.
> Error : exit status 3762504530
> Error ----------------------------------

I think there is a problem with loading assembly sni.dll when starting gauge runtime.

![image](https://user-images.githubusercontent.com/64008859/81428922-b275a400-9165-11ea-92e7-ab0247d1992c.png)

Thanks
Serkan

**Comment by sriv** [comment]
apologies for the delay - is this still an issue?

if so, could you upload the project some place, and I'll try to debug it.

**Comment by sustundag** [comment]
Hi @sriv 

I upgraded sample project latest version and tested again. yes, this is still an issue.
Here you can find the sample project.
[sampleproject](https://drive.google.com/drive/folders/14O8tHcKv9iEGlM81O6y1OgpU0p-BTy7m?usp=sharing)

Thank you for your support
Serkan

**Comment by sriv** [comment]
hi, I can confirm that this is an issue with gauge-dotnet.

I used this sample project - 
[gauge-dotnet-sqlite-example.zip](https://github.com/getgauge/gauge-dotnet/files/4895937/gauge-dotnet-sqlite-example.zip)

This tries to use `EntityFramework` to connect to a sqlite database.

Note that there are two ways to run this project:

- when run using `dotnet run` it runs this project as a standalone .net core app. It throws an exception `Unhandled exception. Microsoft.Data.Sqlite.SqliteException (0x80004005): SQLite Error 1: 'no such table: Products'.` Which indicates that the connection went through.

- when running using `gauge run specs`, the error is different:

```
System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.TypeInitializationException: The type initializer for 'Microsoft.Data.Sqlite.SqliteConnection' threw an exception.
 ---> System.Reflection.TargetInvocationException: Exception has been thrown by the target of an invocation.
 ---> System.DllNotFoundException: Unable to load shared library 'e_sqlite3' or one of its dependencies. In order to help diagnose loading problems, consider setting the LD_DEBUG environment variable: libe_sqlite3: cannot open shared object file: No such file or directory
   at System.Runtime.InteropServices.NativeLibrary.LoadByName(String libraryName, QCallAssembly callingAssembly, Boolean hasDllImportSearchPathFlag, UInt32 dllImportSearchPathFlag, Boolean throwOnError)
   at System.Runtime.InteropServices.NativeLibrary.LoadLibraryByName(String libraryName, Assembly assembly, Nullable`1 searchPath, Boolean throwOnError)
   at System.Runtime.InteropServices.NativeLibrary.Load(String libraryName, Assembly assembly, Nullable`1 searchPath)
   at SQLitePCL.NativeLibrary.Load(String libraryName, Assembly assy, Int32 flags)
   at SQLitePCL.Batteries_V2.MakeDynamic(String name, Int32 flags)
   at SQLitePCL.Batteries_V2.DoDynamic_cdecl(String name, Int32 flags)
   at SQLitePCL.Batteries_V2.Init()
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Microsoft.Data.Sqlite.Utilities.BundleInitializer.Initialize()
   at Microsoft.Data.Sqlite.SqliteConnection..cctor()
   --- End of inner exception stack trace ---
   at Microsoft.Data.Sqlite.SqliteConnection..ctor(String connectionString)
   at Microsoft.EntityFrameworkCore.Sqlite.Storage.Internal.SqliteRelationalConnection.CreateDbConnection()
   at Microsoft.EntityFrameworkCore.Storage.RelationalConnection.get_DbConnection()
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.CreateCommand(RelationalCommandParameterObject parameterObject, Guid commandId, DbCommandMethod commandMethod)
   at Microsoft.EntityFrameworkCore.Storage.RelationalCommand.ExecuteReader(RelationalCommandParameterObject parameterObject)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.InitializeReader(DbContext _, Boolean result)
   at Microsoft.EntityFrameworkCore.Storage.Internal.NoopExecutionStrategy.Execute[TState,TResult](TState state, Func`3 operation, Func`3 verifySucceeded)
   at Microsoft.EntityFrameworkCore.Query.Internal.QueryingEnumerable`1.Enumerator.MoveNext()
   at netcore.template.StepImplementation.PrintProducts() in /home/steam/projects/gauge-dotnet-sqlite-example/StepImplementation.cs:line 17
   --- End of inner exception stack trace ---
   at System.RuntimeMethodHandle.InvokeMethod(Object target, Object[] arguments, Signature sig, Boolean constructor, Boolean wrapExceptions)
   at System.Reflection.RuntimeMethodInfo.Invoke(Object obj, BindingFlags invokeAttr, Binder binder, Object[] parameters, CultureInfo culture)
   at System.Reflection.MethodBase.Invoke(Object obj, Object[] parameters)
   at Gauge.Dotnet.Wrappers.ReflectionWrapper.Invoke(MethodInfo method, Object obj, Object[] args) in /home/steam/projects/gauge-dotnet/src/Wrappers/ReflectionWrapper.cs:line 27
   at Gauge.Dotnet.MethodExecutor.Execute(MethodInfo method, Object[] parameters) in /home/steam/projects/gauge-dotnet/src/MethodExecutor.cs:line 42
   at Gauge.Dotnet.StepExecutor.Execute(GaugeMethod gaugeMethod, String[] args) in /home/steam/projects/gauge-dotnet/src/StepExecutor.cs:line 51
```

This indicates that there is a difference between how gauge-dotnet and dotnet runtime load the assemblies.

Needs more investigatoin. Also worth noting that a lot of gauge-dotnet is a port from `gauge-csharp` which used the .NET framework. So there might be better ways of loading/managing assemblies (expecially given the lack of appdomains in .net core).

**Comment by sriv** [comment]
should be fixed via #106 and released as part of gauge-dotnet v0.1.6. Closing this, please comment if this is still an issue.
